impl/src/main/java/io/kroxylicious/testing/kafka/common/CertificateGenerator.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.testing.kafka.common;
+
+import java.io.FileOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileAttribute;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.security.KeyStore;
+import java.time.Instant;
+import java.util.Set;
+import java.util.UUID;
+
+import io.netty.pkitesting.CertificateBuilder;
+import io.netty.pkitesting.X509Bundle;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+/**
+ * The type Certificate generator.
+ */
+public class CertificateGenerator {
+    private final String domain;
+    private final String ipAddress;
+    private String password;
+    private Path keyStoreFilePath;
+    private KeyStore keyStore;
+
+    /**
+     * Instantiates a new Certificate generator.
+     *
+     * @param domain the domain
+     */
+    public CertificateGenerator(String domain) {
+        this(domain, null);
+    }
+
+    /**
+     * Instantiates a new Certificate generator.
+     * This constructor gives the possibility to add a domain as a SAN dns and
+     * an IP address as a SAN IP.
+     *
+     * @param domain the domain
+     * @param ipAddress the ip address
+     */
+    public CertificateGenerator(String domain, String ipAddress) {
+        this.domain = domain;
+        this.ipAddress = ipAddress;
+    }
+
+    private String getDomainName(String email, String organizationUnit, String organization, String city, String state, String country) {
```

Comment:
The PKI term i, I think,  `Distinguished Name`, see [rfc5280](https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.4) so this should be renamed 
```suggestion
    private String getDistinguishedName(String email, String organizationUnit, String organization, String city, String state, String country) {
```